winml/test/model/model_tests.cpp

```diff
@@ -0,0 +1,166 @@
+#include "testPch.h"
+#include "onnxruntime_cxx_api.h"
+#include "test/onnx/TestCase.h"
+#include "test/onnx/heap_buffer.h"
+#include "test/util/include/test/compare_ortvalue.h"
+#include "ort_value_helper.h"
+
+#ifndef BUILD_GOOGLE_TEST
+#error Must use googletest for value-parameterized tests
+#endif
+
+using namespace onnxruntime::test;
+using namespace winml;
+using namespace onnxruntime;
+
+namespace WinML {
+// Global needed to keep the actual ITestCase alive while the tests are going on. Only ITestCase* are used as test parameters.
+std::vector<std::unique_ptr<ITestCase>> ownedTests;
+
+class ModelTest : public testing::TestWithParam<std::tuple<ITestCase*, winml::LearningModelDeviceKind>> {
+ protected:
+  void SetUp() override {
+    std::tie(m_testCase, m_deviceKind) = GetParam();
+    WINML_EXPECT_NO_THROW(m_testCase->GetPerSampleTolerance(&m_perSampleTolerance));
+    WINML_EXPECT_NO_THROW(m_testCase->GetRelativePerSampleTolerance(&m_relativePerSampleTolerance));
+    WINML_EXPECT_NO_THROW(m_testCase->GetPostProcessing(&m_postProcessing));
+  }
+  winml::LearningModelDeviceKind m_deviceKind;
+  ITestCase* m_testCase;
+  std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> m_converter;
+  double m_perSampleTolerance = 1e-3;
+  double m_relativePerSampleTolerance = 1e-3;
+  bool m_postProcessing = false;
+
+  void BindInputsFromFeed(LearningModelBinding& binding, std::unordered_map<std::string, Ort::Value>& feed) {
+    for (auto it = feed.begin(); it != feed.end(); it++) {
+      auto bindingName = it->first;
+      ITensor bindingValue;
+      WINML_EXPECT_NO_THROW(bindingValue = OrtValueHelpers::LoadTensorFromOrtValue(it->second));
+      WINML_EXPECT_NO_THROW(binding.Bind(m_converter.from_bytes(bindingName), bindingValue));
+    }
```

Comment:
Nit: avoid the unnecessary string copy. You can also use C++17 structured bindings. Something like:

```
for (const auto& [name, value] : feed) {
      ITensor bindingValue;
      WINML_EXPECT_NO_THROW(bindingValue = OrtValueHelpers::LoadTensorFromOrtValue(value));
      WINML_EXPECT_NO_THROW(binding.Bind(m_converter.from_bytes(bindingName), bindingValue));
}
```

Reply:
Fixed